src/pages/ProductsCatalog/ProductsCatalog.tsx

```diff
@@ -231,12 +235,23 @@ export const ProductsCatalog: FC<ProductsCatalogProps> = ({ category }) => {
             value={showItemsPerPageDropdownValue()}
             onChange={value => {
               if (value) {
-                setSearchParams(currentParams => {
-                  const itemsParams = value.value;
+                const newCountOfPages = Math.ceil(
+                  sortedPhones.length / value.value,
+                );
+                const itemsParams = value.value;
+                const newParams =
+                  newCountOfPages < currentPage
+                    ? {
+                        devicesPerPage: itemsParams.toString(),
+                        page: newCountOfPages.toString(),
+                      }
+                    : {
+                        devicesPerPage: itemsParams.toString(),
+                        page: currentPage.toString(),
+                      };
```

Comment:
You can do it withou duplication of code parts

const newParams = {
  devicesPerPage: itemsPerPage.toString(),
  page: (newCountOfPages < currentPage ? newCountOfPages : currentPage).toString(),
};